src/soc/soc.c

```diff
@@ -59,6 +59,14 @@ void soc_register(struct soc_t **soc, char *brand, char *type) {
 	(*soc)->selectableFd = NULL;
 	(*soc)->gc = NULL;
 
+	(*soc)->I2CSetup = NULL;
```

Comment:
If you're not using them, then remove them altogether. You introduced these members yourself.

Reply:
I have used them in the end of the file rk3399.c

Reply:
Again, why? Since your not exposing them outside of the rk3399.c file since your not interfacing them anywhere.

Reply:
I'm sorry that did you mean I should add  function  like "EXPORT int I2CSetup(int devFd)" if I want to add (*soc)->I2CSetup ?

Reply:
I'm sorry that did you mean I should add function like "EXPORT int I2CSetup(int devFd)" in wiringX.c if I want to add (*soc)->I2CSetup in soc.c ?

Reply:
I've learned that I can call I2C functions directly，rather than add (*soc)->I2CSetup in soc.c. Right？

Reply:
Since you removed all I2C functions, you can now also remove the members you added to the struct.